docs/001-mana_proposal.md

```diff
@@ -0,0 +1,469 @@
+ - `update(nodeID, time)`: update `Base Mana 2`, `Effective Base Mana 1` and `Effective Base Mana 2` of a node with respect `time`.
+ - `updateAll(time)`: update `Base Mana 2`, `Effective Base Mana 1` and `Effective Base Mana 2` of all nodes with respect to `time`.
+
+`BaseMana` should have the following methods:
+ - `pledgeAndUpdate(amount, time)`: update `BaseMana` fields with respect to `time` and pledge `amount` mana.
+ - `revokeBaseMana1(amount, time)`:  update `BaseMana` values with respect to `time` and revoke `amount` `BaseMana1`.
+ - `update(time)`: update all `BaseMana` fields with respect to `time`.
+
+#### Base Mana Calculation
+
+There are two cases when the values within `Base Mana Vector` are updated:
+ 1. A confirmed transaction pledges mana.
+ 2. Any module accesses the `Base Mana Vector`, and hence its values are updated with respect to `access time`.
+
+First, let's explore the former.
+
+##### A confirmed transaction pledges mana
+For simplicity, we only describe mana calculation for one of the Base Mana Vectors, namely, the Base Access Mana Vector.
+
+First, a `TransactionConfirmed` event is triggered, therefore `BaseManaVector.BookMana(transaction)` is executed:
+```go
+func (bmv *BaseManaVector) BookMana(tx transaction) {
+    t := tx.timestamp
+    pledgedNodeID := tx.accessMana
+
+    amount := sumBalances(tx.outputs)
+
+    for input := range tx.inputs {
+        // search for the nodeID that the input's tx pledged its mana to
+        inputNodeID := loadPledgedNodeIDFromInput(input)
+        bmv[inputNodeID].revokeBaseMana1(input.balance, t)
+        Events.ManaRevoked.Trigger(&ManaRevokedEvent{inputNodeID, input.balance, bmv[inputNodeID], AccessManaType})
+    }
+
+    bmv[pledgedNodeID].pledgeAndUpdate(amount, time)
+    Events.ManaPledged.Trigger(&ManaPledgedEvent{pledgedNodeID, amount, bmv[pledgedNodeID], AccessManaType})
+}
+```
+`Base Mana 1` is being revoked from the nodes that pledged mana for inputs that the current transaction consumes.
+Then, the appropriate node is located in `Base Mana Vector`, and sum of the transaction outputs amount of mana is pledged to it's `BaseMana`.
+
+Note, that `revokeBaseMana1` accesses the mana entry of the nodes within `Base Mana Vector`, therefore all values are updated with respect to `t`.
+```go
+func (bm *BaseMana) revokeBaseMana1(amount float64, t time.Time) {
+    n := t - bm.LastUpdated
+    // first, update EBM1, BM2 and EBM2 until `t`
+    bm.updateEBM1(n)
+    bm.updateBM2(n)
+    bm.updateEBM2(n)
+
+    bm.LastUpdated = t
+    // revoke BM1 at `t`
+    bm.BaseMana1 -= amount
+}
+```
+
+```go
+func (bm *BaseMana) pledgeAndUpdate(amount float64, t time.Time) {
+    n := t - bm.LastUpdated
+    // first, update EBM1, BM2 and EBM2 until `t`
+    bm.updateEBM1(n)
+    bm.updateBM2(n)
+    bm.updateEBM2(n)
+
+    bm.LastUpdated = t
+
+    bm.BaseMana1 += amount
+    bm.BaseMana2 += amount
```

Comment:
The base Mana 2 does not get updated by the whole amount, it uses the "pending mana", it's a formula that depends also on how long the funds sat on an address. So it would look like this:

for input := range tx.inputs {
         // search for the timestamp of the UTXO that generated the input
         t_inp :=  input.timestamp (??? This is probably wrong, but you know what I mean...)
         bm.BaseMana2 += scale/decay*input.balance*(1-math.Pow( math.E,-decay*(tx.timestamp-t_inp) ) ) 
     }

Reply:
Good point! I didn't take into account the time varying nature of pending mana. In your comment, you assume that `alpha` = 1, right? Then if I plot the graph with the `gamma` value from the specs, I get this function:

![image](https://user-images.githubusercontent.com/32927267/91705112-d037a080-eb7c-11ea-85d8-e5fba98ff75f.png)

So it's far to say, that pending mana with these parameters reaches its upper bound in appr. 3000 seconds, which is 50 minutes?